app/routes/_layout+/settings.custom-fields.index.tsx

```diff
@@ -0,0 +1,119 @@
+import { OrganizationType, type CustomField } from "@prisma/client";
+import { json, redirect } from "@remix-run/node";
+import type { ActionArgs, LoaderArgs, V2_MetaFunction } from "@remix-run/node";
+import { ActionsDropdown } from "~/components/custom-fields/actions-dropdown";
+import { ErrorBoundryComponent } from "~/components/errors";
+import type { HeaderData } from "~/components/layout/header/types";
+import { List } from "~/components/list";
+import { Button } from "~/components/shared/button";
+import { Td } from "~/components/table";
+import { db } from "~/database";
+import { requireAuthSession } from "~/modules/auth";
+import { getCustomFields } from "~/modules/custom-field";
+import { getOrganizationByUserId } from "~/modules/organization";
+import {
+  getCurrentSearchParams,
+  getParamsValues,
+  generatePageMeta,
+} from "~/utils";
+import { appendToMetaTitle } from "~/utils/append-to-meta-title";
+
+export const meta: V2_MetaFunction<typeof loader> = ({ data }) => [
+  { title: data ? appendToMetaTitle(data.header.title) : "" },
+];
+
+export const ErrorBoundary = () => <ErrorBoundryComponent />;
+
+export async function loader({ request }: LoaderArgs) {
+  const { userId } = await requireAuthSession(request);
+  const searchParams = getCurrentSearchParams(request);
+  const { page, perPage, search } = getParamsValues(searchParams);
+  const { prev, next } = generatePageMeta(request);
+  const organization = await getOrganizationByUserId({
+    userId,
+    orgType: OrganizationType.PERSONAL,
+  });
+
+  if (!organization) {
+    throw new Error("Organization not found");
+  }
+
+  const { customFields, totalCustomFields } = await getCustomFields({
+    organizationId: organization.id,
+    page,
+    perPage,
+    search,
+  });
+  const totalPages = Math.ceil(totalCustomFields / perPage);
+
+  const header: HeaderData = {
+    title: "Custom Fields",
+  };
+  const modelName = {
+    singular: "custom fields",
+    plural: "custom Fields",
+  };
+  return json({
+    header,
+    items: customFields,
+    search,
+    page,
+    totalItems: totalCustomFields,
+    totalPages,
+    perPage,
+    prev,
+    next,
+    modelName,
+  });
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  await requireAuthSession(request);
```

Comment:
Here we only have a delete action so I would use the `isDelete()` helper to make sure the request has the correct method.